pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -211,6 +213,23 @@ func (cs *ConntrackConnectionStore) AddOrUpdateConn(conn *flowexporter.Connectio
 		existingConn.ReverseBytes = conn.ReverseBytes
 		existingConn.ReversePackets = conn.ReversePackets
 		existingConn.TCPState = conn.TCPState
+
+		prevIsActive := existingConn.PriorityQueueItem.IsActive
+		// Reset the pqItem idle expire time and set IsActive to be true when
+		// there are changes either in stats or TCP state.
+		if (conn.OriginalPackets > existingConn.PriorityQueueItem.PrevPackets) ||
+			(conn.ReversePackets > existingConn.PriorityQueueItem.PrevReversePackets) ||
+			(existingConn.PriorityQueueItem.Conn.TCPState != conn.TCPState) {
+			existingConn.PriorityQueueItem.IsActive = true
+			existingConn.PriorityQueueItem.IdleExpireTime = time.Now().Add(cs.ConnectionStore.ExpirePriorityQueue.IdleFlowTimeout)
+		}
+		// We need to update the connection value copy stored in the pqItem.
+		if prevIsActive {
+			cs.ConnectionStore.ExpirePriorityQueue.Update(existingConn.PriorityQueueItem, connKey, existingConn,
+				existingConn.PriorityQueueItem.ActiveExpireTime, existingConn.PriorityQueueItem.IdleExpireTime)
+		} else {
+			cs.ConnectionStore.addItemToQueue(connKey, conn, false)
```

Comment:
Also, is it necessary to add this item immediately? Why don't we add this only when the connection becomes active again, because there will be cases where the connection stays idle for multiple poll cycles.

Reply:
you're right, I thought I was adding the it back only when idle item becomes active again. will correct this